### PR TITLE
Add ability to set a threshold per key

### PR DESCRIPTION
**Pull request by tikotzky**
**What**:
Add the ability to set a threshold per key

**Why**:
To be able to set specific keys to only match when they meet a specified threshold

**How**:
By adding returning the keys threshold with each match and verifying that the the match is `>=` the threshold specified on its key.

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Ready to be merged 
* [x] Added myself to contributors table

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -60,17 +65,21 @@ function matchSorter(items, value, options = {}) {
  * @param {Array} keys - the keys to get values from the item for the ranking
  * @param {String} value - the value to rank against
  * @param {Object} options - options to control the ranking
- * @return {{rank: Number, keyIndex: Number}} - the highest ranking
+ * @return {{rank: Number, keyIndex: Number, itemThreshold: Number}} - the highest ranking
```
Comment (kentcdodds) [inline src/index.js]: Should we call this `keyThreshold` instead?

Comment (tikotzky) [reply src/index.js]: Yup. Agree. I’ll make the change soon.

**Comment by tikotzky** [comment]
Thanks!

**Pull request merged**